pkg/k8s/apis/cilium.io/client/register_v1_crd.go

```diff
@@ -76,44 +79,61 @@ func GetPregeneratedCRD(crdName string) apiextensionsv1.CustomResourceDefinition
 	scopedLog := log.WithField("crdName", crdName)
 
 	switch crdName {
+	// Tracing Policy Custom Resource Definition.
 	case v1alpha1.TPCRDName:
 		crdBytes = crdsv1Alpha1TracingPolicies
+	// Namespaced Tracing Policy Custom Resource Definition.
 	case v1alpha1.TPNamespacedCRDName:
 		crdBytes = crdsv1Alpha1TracingPoliciesNamespaced
+	// Pod Info Custom Resource Definition.
+	case v1alpha1.PICRDName:
+		crdBytes = crdsv1Alpha1PodInfo
 	default:
-		scopedLog.Fatal("Pregenerated CRD does not exist")
+		scopedLog.Fatal("Pre-generated CRD does not exist")
 	}
 
 	isoCRD := apiextensionsv1.CustomResourceDefinition{}
 	err = yaml.Unmarshal(crdBytes, &isoCRD)
 	if err != nil {
-		scopedLog.WithError(err).Fatal("Error unmarshalling pregenerated CRD")
+		scopedLog.WithError(err).Fatal("Error unmarshalling pre-generated CRD")
 	}
 
 	return isoCRD
 }
 
-func createTPCRDs(clientset apiextensionsclient.Interface) error {
-	// custer-wide tracing policy CRD
-	isoCRD := GetPregeneratedCRD(v1alpha1.TPCRDName)
-	if err := createUpdateCRD(
-		clientset,
-		v1alpha1.TPCRDName,
-		constructV1CRD(v1alpha1.TPName, isoCRD),
-		newDefaultPoller(),
-	); err != nil {
-		return err
+// createTPCRDs creates or updates the CRDs with the API server.
+func createCRDs(clientset apiextensionsclient.Interface, skipPodInfo bool) error {
+
+	doCreateCRD := func(crdName, name string) error {
+		isoCRD := GetPregeneratedCRD(crdName)
+		err := createUpdateCRD(
+			clientset,
+			crdName,
+			constructV1CRD(name, isoCRD),
+			newDefaultPoller())
+		if err != nil {
+			err = fmt.Errorf("failed to create %s: %w", crdName, err)
+			return err
+		}
+		return nil
 	}
 
-	// namespaced tracing policy CRD
-	isoCRD = GetPregeneratedCRD(v1alpha1.TPNamespacedCRDName)
-	if err := createUpdateCRD(
-		clientset,
-		v1alpha1.TPNamespacedCRDName,
-		constructV1CRD(v1alpha1.TPNamespacedName, isoCRD),
-		newDefaultPoller(),
-	); err != nil {
-		return err
+	var ret error
+	// Create TracingPolicy Custom Resource Definition
+	ret = goerrors.Join(ret, doCreateCRD(v1alpha1.TPCRDName, v1alpha1.TPName))
+
+	// Create TracingPoliciesNamespaced Custom Resource Definition
+	ret = goerrors.Join(ret, doCreateCRD(v1alpha1.TPNamespacedCRDName, v1alpha1.TPNamespacedName))
+
+	if skipPodInfo {
+		return ret
+	}
+
+	// Create the PodInfo Custom Resource Definition.
+	ret = goerrors.Join(ret, doCreateCRD(v1alpha1.PICRDName, v1alpha1.PIName))
+
+	if goerrors.Unwrap(ret) != nil {
+		return ret
```

Comment:
Is the `Unwrap` needed?
Docs [say](https://pkg.go.dev/errors#Join): 
> Join returns an error that wraps the given errors. Any nil error values are discarded. Join returns nil if every value in errs is nil.

Reply:
I don't think it is needed, since `ret` will either be nil, or non-nil. I was not very sure about the possibilities so used unwrap there.